tools/runners/run-mongo.js

```diff
@@ -613,10 +632,16 @@ var launchMongo = function (options) {
       }
 
       try {
-        var initiateResult = yieldingMethod(
-          db.admin(), 'command', {replSetInitiate: configuration});
+        yieldingMethod(db.admin(), 'command', {
+          replSetInitiate: configuration,
+        });
       } catch (e) {
-        if (e.message !== 'already initialized') {
+        if (e.message === 'already initialized') {
+          yieldingMethod(db.admin(), 'command', {
+            replSetReconfig: configuration,
+            force: true,
```

Comment:
Without `force: true`, the exception indicates that no changes were made against the secondary (because it wasn't primary).